### PR TITLE
Setting screen

### PR DESCRIPTION
**Pull request by cp-ishita-g**
- [x] design setting screen
- [x] design profile screen
- [x] design edit space screen
- [x] design contact support screen
- [x] create contact support service file
- [x] manage delete and leave space, delete and sign out account
- [x] manage cloud function to send support report

edit spaceedit profile and delete accountcontact support

### CODE REVIEW

File: app/lib/ui/flow/setting/setting_screen.dart [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+class _SettingScreenState extends ConsumerState<SettingScreen> {
+  late SettingViewNotifier notifier;
+
+  @override
+  void initState() {
+    super.initState();
+    runPostFrame(() {
+      notifier = ref.watch(settingViewStateProvider.notifier);
+      notifier.getUserSpace();
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/setting/setting_screen.dart:34]: Let's make this call from notifier constuctor

File: app/lib/ui/flow/setting/setting_view_model.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  }
+
+  void signOut() async {
+    try {
+      state = state.copyWith(signingOut: true);
+      await userService.signOut();
+      state = state.copyWith(signingOut: false, logOut: true);
+    } catch (error, stack) {
+      logger.e(
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/setting/setting_view_model.dart:62]: Show appropriate message to user on error, do changes wherever require

File: data/lib/api/auth/api_user_service.dart [hunk 3]
```diff
@@ -91,6 +104,16 @@ class ApiUserService {
     return null;
   }
 
+  Stream<ApiUser> getUserStream(String userId) {
+    return _userRef.doc(userId).snapshots().map((snapshot) {
+      if (snapshot.exists) {
+        return snapshot.data() as ApiUser;
+      } else {
+        throw Exception('User not found');
```
Comment (cp-radhika-s) [inline data/lib/api/auth/api_user_service.dart:112]: Let's not throw an exception here, it's the caller's responsibility to handle the null data or failure

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/lib/ui/flow/setting/contact_support/contact_support_screen.dart [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+  Widget _submitButton(BuildContext context, ContactSupportViewState state) {
+    return BottomStickyOverlay(
+      child: PrimaryButton(
+        enabled: !state.submitting && state.enableSubmit,
+        progress: state.submitting,
+        context.l10n.contact_support_submit_title,
+        onPressed: () {
+          notifier.submitSupportRequest();
+          showErrorSnackBar(context, state.error.toString());
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/setting/contact_support/contact_support_screen.dart:197]: which error we might have here?

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/lib/ui/flow/setting/contact_support/contact_support_screen.dart (hunk 4 above)
Comment (cp-ishita-g) [reply app/lib/ui/flow/setting/contact_support/contact_support_screen.dart:197]: it was mistake, i am testing this

File: app/lib/ui/flow/setting/space/edit_space_view_model.dart [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        spaceName: TextEditingController(text: space?.space.name),
+        loading: false,
+      );
+    } catch (error, stack) {
+      logger.e(
+        'EditSpaceViewNotifier: error while fetch space details',
+        error: error,
+        stackTrace: stack
+      );
+      state = state.copyWith(error: error);
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/setting/space/edit_space_view_model.dart:53]: set loading false on error, do changes wherever required

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

File: app/lib/ui/flow/setting/contact_support/contact_support_screen.dart [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  }
+
+  Widget _submitButton(BuildContext context, ContactSupportViewState state) {
+    return BottomStickyOverlay(
+      child: PrimaryButton(
+        enabled: !state.submitting && state.enableSubmit,
+        progress: state.submitting,
+        context.l10n.contact_support_submit_title,
+        onPressed: () {
+          notifier.submitSupportRequest();
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/setting/contact_support/contact_support_screen.dart:196]: We need to show a success message on request submitted

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**